doc/source/whatsnew/v1.0.0.rst

```diff
@@ -439,6 +439,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrameGroupby.agg` not able to use lambda function with named aggregation (:issue:`27519`)
 - Bug in :meth:`DataFrame.groupby` losing column name information when grouping by a categorical column (:issue:`28787`)
 - Bug in :meth:`DataFrameGroupBy.rolling().quantile()` ignoring ``interpolation`` keyword argument (:issue:`28779`)
+- Bug in :meth:`DataFrameGroupBy.agg` with timezone-aware datetime64 column incorrect casting results to the original dtype (:issue:`29641`)
```

Comment:
can you add a sub-section in api-changes. this is a non-trivial change i think.

Reply:
sure, but since this note is orthogonal to this PR, im going to do that in a separate PR.  removing this note for now.

Reply:
k